### PR TITLE
CI with mamba

### PR DESCRIPTION
**Pull request by kfir4444**
Work In Progress, trying to make CI work with mamba. Please do not merge.

### CODE REVIEW

**Comment by alongd** [comment]
try `self.assertTrue(almost_equal_coords_lists(new_xyz, expected_xyz))`

File: arc/species/zmatTest.py [hunk 1]
```diff
@@ -1151,7 +1151,8 @@ def test_zmat_to_coords(self):
                                    (-3.871932295381926, -0.9306597576029849, -2.5400987960833166),
                                    (-3.440082669618663, 0.6565731451487455, -1.8462094217817948)]
         expected_ordered_symbols = ['C', 'C', 'H', 'H', 'H', 'H', 'H', 'H', 'C', 'C', 'O', 'O', 'H', 'H', 'H', 'H', 'H']
-        self.assertEqual(ordered_coords, expected_ordered_coords)
+        for coord1,coord2 in zip(expected_ordered_coords,ordered_coords):
+            self.assertAlmostEqual(coord1,coord2)
```
Comment (alongd) [inline arc/species/zmatTest.py:1155]: Replace this (and in L1666) with
`self.assertTrue(almost_equal_coords_lists(coord1, coord2))`

Of course, you'll need to do add at the top: `from arc.common import almost_equal_coords_lists`

File: devtools/install_autotst.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# temporarily change directory to install software, and move one directory up in the tree
```
Comment (alongd) [inline devtools/install_autotst.sh:1]: install_autotst.sh shouldn't be part of this commit

File: .github/workflows/cont_int.yml [hunk 3]
```diff
@@ -70,7 +70,7 @@ jobs:
 
     - name: Lint with flake8
       run: |
-        conda install flake8
+        mamba install flake8
```
Comment (alongd) [inline .github/workflows/cont_int.yml:73]: I think we should squash all cont_int.yml commits together

File: arc/species/converterTest.py [hunk 4]
```diff
@@ -4248,7 +4248,7 @@ def test_modify_coords(self):
         new_xyz = converter.modify_coords(coords=xyz1, indices=indices, new_value=new_val,
                                           modification_type=modification_type, mol=mol1)
         self.assertTrue(almost_equal_coords_lists(new_xyz, expected_xyz))
-        self.assertAlmostEqual(converter.get_zmat_param_value(coords=new_xyz, indices=indices, mol=mol1), new_val)
+        self.assertAlmostEqual(converter.get_zmat_param_value(coords=new_xyz, indices=indices, mol=mol1), new_val, 5)
```
Comment (alongd) [inline arc/species/converterTest.py:4251]: Please change the commit message, currently it says "Merge branch 'main' into CI_with_mamba "

**Review by alongd — COMMENTED** [review]
Congrats that this branch passes all tests!!
It will be interesting to try re-running the tests and see whether the cache works now and it takes less time. Currently it takes ~75 min (used to be ~140 min) to run CI tests online.
I added some comments, please take a look

**Comment by kfir4444** [comment]
@alongd it seems that the CI/build test on PR is failing, due to an issue with a test marked with `@work_in_progress`. I will try to push again and see if the problem returns, but if not, we might need to consider this PR, especially with the warning [here](https://github.com/ReactionMechanismGenerator/ARC/actions/runs/2263993967) and [here](https://github.com/ReactionMechanismGenerator/ARC/actions/runs/2263994060).

**Comment by alongd** [comment]
The problem seems to be with GCN, I don't think it's an issue in the current PR, I'm having GCN problems as well with the Job PR #504.

File: .github/workflows/cont_int.yml [hunk 5]
```diff
@@ -1,11 +1,18 @@
...
+      - '*'
   pull_request:
     branches:
       - main
     types: [opened, synchronize, reopened, ready_for_review, review_requested]
+  schedule:
+    - cron: '0 0 * * *'
+
+env:
+  CACHE_NUMBER: 1  # increase to reset conda env cache manually
```
Comment (alongd) [inline .github/workflows/cont_int.yml:15]: This should be 0... If the test only passes (once) with a value of 1, it means that the cache probably doesn't work.
Before merging, we should report how long (approximately) it takes to run a test for the first time (i.e., changing `CACHE_NUMBER` to some random number to have a new cache), and how long it takes to run it for the 2nd time with the same `CACHE_NUMBER` value by restarting the test.
If the test repeatedly fails for the second time per `CACHE_NUMBER` value, then should stop using the cache (modify this file) if we cannot solve it, the mamba speed-up is already a nice feature.

File: .github/workflows/cont_int.yml [hunk 6]
```diff
@@ -17,16 +24,30 @@ jobs:
...
-        auto-activate-base: false
+        use-mamba: true
+
+    - name: Set cache date
+      run: echo "DATE=$(date +'%Y%m%d')" >> $GITHUB_ENV
+
+    - uses: actions/cache@v2
+      with:
+        path: /usr/share/miniconda3/envs/my-env
+        key: linux-64-conda-${{ hashFiles('environment.yml') }}-${{ env.DATE }}-${{ env.CACHE_NUMBER }}
```
[the remark this answers is not in the document]
Comment (alongd) [reply .github/workflows/cont_int.yml:42]: Can you take a look at the comment above?

File: .github/workflows/cont_int.yml [hunk 7]
```diff
@@ -36,6 +57,7 @@ jobs:
         git clone https://github.com/ReactionMechanismGenerator/RMG-Py
         cd RMG-Py
         export PYTHONPATH=$PYTHONPATH:$(pwd)
+        conda install -c anaconda cython
```
Comment (alongd) [inline .github/workflows/cont_int.yml:60]: conda or mamba?

**Review by alongd — COMMENTED** [review]
Thanks for taking care of this PR! Please see some comments below. Feel free to squash the commits.

**Review by alongd — APPROVED** [review]
Looking great, thanks for taking the time to sort this out!
Crossing figures that rebasing `job` on top of this will do the trick :)
